player.go

```diff
@@ -143,23 +180,39 @@ func (p *Player) Play(uri *string) error {
 	if err := p.Player.Load(track); err != nil {
 		return err
 	}
-
+	log.Println("Track", )
 	// Defer unloading the track until we exit this func
 	defer p.Player.Unload()
 
 	// Play the track
 	log.Println(fmt.Sprintf("Playing: %s", *uri))
 	p.Player.Play() // This does NOT block, we must block ourselves
+	p.TrackTicker.Start()
+
+	// Runs a loop which increases track duration every second
+	go func() (error) {
+		for {
+			tick := time.Tick(1 * time.Second)
+			select {
+				case <-StopTimer:
+					return nil
+				case <-tick:
+					p.TrackTicker.Increase()
+			}
+		}
+	}()
 
 	// Go routine to listen for end of track updates from the player, once we get one
 	// send a message to our own StopTrack channel
 	go func() {
 		<-p.Session.EndOfTrackUpdates() // Blocks
 		log.Println("End of Track Updates - Stop Track")
+		p.TrackTicker.Stop()
 		StopTrack <- struct{}{}
 	}()
```

Comment:
Might be cool to put this one into its one method too whilst you are at it :wink: